### PR TITLE
[FIXED] private import issue by pulling in up to date jwt library

### PR DESCRIPTION
**Pull request by matthiashanel**
Also prevent nats based account resolver from storing invalid jwt
    Updated compress and highwayhash

Signed-off-by: Matthias Hanel <mh@synadia.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 
-go 1.15
+go 1.16
 
 require (
-	github.com/klauspost/compress v1.11.7
-	github.com/minio/highwayhash v1.0.0
-	github.com/nats-io/jwt/v2 v2.0.0-20210208203759-ff814ca5f813
+	github.com/klauspost/compress v1.11.12
+	github.com/minio/highwayhash v1.0.1
```
Comment (wallyqs) [inline go.mod:7]: This patch version bump seems mostly updates to run in graviton so should be ok: https://github.com/minio/highwayhash/compare/v1.0.0...v1.0.1

File: go.mod [hunk 2]
```diff
@@ -1,15 +1,15 @@
 module github.com/nats-io/nats-server/v2
 
-go 1.15
+go 1.16
 
 require (
-	github.com/klauspost/compress v1.11.7
-	github.com/minio/highwayhash v1.0.0
-	github.com/nats-io/jwt/v2 v2.0.0-20210208203759-ff814ca5f813
+	github.com/klauspost/compress v1.11.12
```
Comment (wallyqs) [inline go.mod:6]: https://github.com/klauspost/compress/compare/v1.11.7...v1.11.12

File: go.mod (hunk 1 above)
Comment (matthiashanel) [reply go.mod:7]: since I updated jwt, we bumped these as well

Comment (kozlovic) [reply go.mod:7]: We decided to update all deps since this is a new release.

**Review by kozlovic — APPROVED** [review]
LGTM

File: vendor/modules.txt [hunk 3]
```diff
@@ -1,30 +1,30 @@
-# github.com/klauspost/compress v1.11.7
+# github.com/klauspost/compress v1.11.12
 ## explicit
 github.com/klauspost/compress/s2
-# github.com/minio/highwayhash v1.0.0
+# github.com/minio/highwayhash v1.0.1
 ## explicit
 github.com/minio/highwayhash
-# github.com/nats-io/jwt/v2 v2.0.0-20210208203759-ff814ca5f813
+# github.com/nats-io/jwt/v2 v2.0.1
```
Comment (wallyqs) [inline vendor/modules.txt:7]: Might be my machine but maybe we are missing jwt v1 here (which jwt/v2 imports internally), I got this locally after running `go mod vendor`:

```diff
+# github.com/nats-io/jwt v1.2.2
+github.com/nats-io/jwt
+# github.com/nats-io/jwt/v2 v2.0.1
 ## explicit
 github.com/nats-io/jwt/v2
```

Comment (kozlovic) [reply vendor/modules.txt:7]: I don't, but doing a `go mod tidy` does update the go.sum...

Comment (kozlovic) [reply vendor/modules.txt:7]:
```

**Pull request merged**